library/core/src/task/wake.rs

```diff
@@ -531,12 +531,26 @@ impl Waker {
 
     /// Returns a reference to a `Waker` that does nothing when used.
     ///
+    // Note!  Much of the documentation for this method is duplicated
+    // in the docs for `LocalWaker::noop`.
+    // If you edit it, consider editing the other copy too.
+    //
     /// This is mostly useful for writing tests that need a [`Context`] to poll
     /// some futures, but are not expecting those futures to wake the waker or
     /// do not need to do anything specific if it happens.
     ///
     /// If an owned `Waker` is needed, `clone()` this one.
     ///
+    /// # Warning
+    ///
+    /// Do not use this function as a way to be able to call `poll` methods,
+    /// in production code,
```

Comment:
If we're using semantic linefeeds then let's actually use the 100 characters we're allowed in our doc texts. Also, these commas are redundant.
```suggestion
    /// Do not use this function as a way to be able to call `poll` methods in production code
```